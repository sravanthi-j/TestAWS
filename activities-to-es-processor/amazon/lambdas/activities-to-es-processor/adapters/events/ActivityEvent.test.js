const expect = require('expect');
const ActivityEvent = require('./ActivityEvent');

test('When event is populated, can create ES message', () => {
  const id = "MyId";
  const timestamp = new Date('2019-12-17T03:24:00Z');
  const index = "MyIndex"; //TODO this probably comes from config

  const event = new ActivityEvent(id, timestamp, { id: id, timestamp: timestamp, foo: 1});
  const result = event.toElasticSearchFormat(index);

  expect(result).toMatchSnapshot();
});