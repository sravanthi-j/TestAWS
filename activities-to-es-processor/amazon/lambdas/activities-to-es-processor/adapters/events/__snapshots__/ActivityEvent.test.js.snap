// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When event is populated, can create ES message 1`] = `
"{\\"index\\":{\\"_index\\":\\"MyIndex\\",\\"_id\\":\\"MyId\\"}}
{\\"@id\\":\\"MyId\\",\\"@timestamp\\":\\"2019-12-17T03:24:00.000Z\\",\\"@message\\":{\\"id\\":\\"MyId\\",\\"timestamp\\":\\"2019-12-17T03:24:00.000Z\\",\\"foo\\":1}}
"
`;
